src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1673,12 +1676,29 @@ RouterJob(Query *originalQuery, PlannerRestrictionContext *plannerRestrictionCon
  * By default it does not reorder the task list, implying a first-replica strategy.
  */
 static void
-ReorderTaskPlacementsByTaskAssignmentPolicy(Job *job, TaskAssignmentPolicyType
-											taskAssignmentPolicy)
+ReorderTaskPlacementsByTaskAssignmentPolicy(Job *job,
+											TaskAssignmentPolicyType taskAssignmentPolicy,
+											List *placementList)
 {
 	if (taskAssignmentPolicy == TASK_ASSIGNMENT_ROUND_ROBIN)
 	{
-		job->taskList = RoundRobinAssignTaskList(job->taskList);
+		ShardPlacement *primaryPlacement = NULL;
+
+		/*
+		 * We hit a single shard on router plans, and there should be only
+		 * one task in the TaskList
+		 * */
```

Comment:
minor: last line of comment needs formatting.

Reply:
I believe it is ok now